src/utils/hooks/transaction/utils/fee.ts

```diff
@@ -0,0 +1,119 @@
+import {
+  CurrencyExt,
+  CurrencyId,
+  ExtrinsicData,
+  isCurrencyEqual,
+  LiquidityPool,
+  MultiPath,
+  newCurrencyId,
+  Trade
+} from '@interlay/interbtc-api';
+import { MonetaryAmount } from '@interlay/monetary-js';
+import { SubmittableExtrinsic } from '@polkadot/api/types';
+
+import { GOVERNANCE_TOKEN } from '@/config/relay-chains';
+
+import { getExtrinsic } from '../extrinsics';
+import { TransactionActions } from '../types';
+
+const constructSwapPathPrimitive = (path: MultiPath): Array<CurrencyId> => {
+  const inputCurrency = newCurrencyId(window.bridge.api, path[0].input);
+  return [inputCurrency, ...path.map(({ output }) => newCurrencyId(window.bridge.api, output))];
+};
+
+// Recursively double input amount until the trade with higher than minimum output
+// amount is found.
+const getOptimalTradeForTxFeeSwap = (
+  minOutputAmount: MonetaryAmount<CurrencyExt>,
+  inputAmount: MonetaryAmount<CurrencyExt>,
+  pools: Array<LiquidityPool>
+): Trade => {
+  const trade = window.bridge.amm.getOptimalTrade(inputAmount, minOutputAmount.currency, pools);
+
+  if (trade === null || trade.outputAmount.lt(minOutputAmount)) {
+    // If the output amount is lower than minimum trade or minimum txFee
+    // then double the input currency amount and check again.
+    return getOptimalTradeForTxFeeSwap(minOutputAmount, inputAmount.mul(2), pools);
+  }
+  return trade;
+};
+
+const getTxFeeSwapData = async (
+  nativeTxFee: MonetaryAmount<CurrencyExt>,
+  feeCurrency: CurrencyExt,
+  baseExtrinsic: SubmittableExtrinsic<'promise'>,
+  pools: Array<LiquidityPool>
+): Promise<{ swapPathPrimitive: Array<CurrencyId>; inputAmount: MonetaryAmount<CurrencyExt> }> => {
+  // First we construct reverse direction trade to get estimated swap path and amount
+  const reverseDirectionTrade = window.bridge.amm.getOptimalTrade(nativeTxFee, feeCurrency, pools);
+  if (reverseDirectionTrade === null) {
+    throw new Error(
+      `Not possible to exchange ${feeCurrency.name} for ${nativeTxFee.currency.name}: trade path not found.`
+    );
+  }
+  // Final native token transaction fee is estimated for base extrinsic wrapped in multiTransactionPayment call.
+  // NOTE: We assume here the reverse direction trade has similar weight.
+  const reverseDirectionExtrinsic = window.bridge.api.tx.multiTransactionPayment.withFeeSwapPath(
+    constructSwapPathPrimitive(reverseDirectionTrade.path),
+    reverseDirectionTrade.outputAmount.toString(true),
+    baseExtrinsic
+  );
+  const withSwapTxFee = await window.bridge.transaction.getFeeEstimate(reverseDirectionExtrinsic);
+  const { inputAmount, path } = getOptimalTradeForTxFeeSwap(
+    withSwapTxFee.mul(1.5), // 50% on top to be safe
```

Comment:
Put the 1.5 as a constant at the top of the file so it's easy to see instead of having the magic number in the logic.